flutter/lib/src/widgets_binding_observer.dart

```diff
@@ -0,0 +1,155 @@
+class SentryWidgetsBindingObserver with WidgetsBindingObserver {
+  SentryWidgetsBindingObserver({Hub hub}) {
+    this.hub = hub ?? HubAdapter();
+  }
+
+  Hub hub;
+
+  /// This method records lifecycle events.
+  /// It tries to mimic the behavior of ActivityBreadcrumbsIntegration of Sentry
+  /// Android for lifecycle events.
+  ///
+  /// On Android and iOS this records lifecycle event breadcrumbs which loosely
+  /// correspond to their respectiv platforms lifecycle events.
+  ///
+  /// Does this need to be tracked? The nativ Android an iOS integration
+  /// also tracks this but what about Web, Linux, Windows and MacOS?
+  /// The message could include that this is a Flutter message.
+  ///
+  /// See also:
+  ///   - [WidgetsBindingObserver](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html)
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    // According to
+    // https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
+    // this is of the type navigation.
+    hub.addBreadcrumb(Breadcrumb(
+      message: _lifecycleToString(state),
+      category: 'ui.lifecycle',
+      type: 'navigation',
+    ));
+  }
+
+  /// Called when the application's dimensions change. For example,
+  /// when a phone is rotated or an application window is resized.
+  ///
+  /// See also:
+  ///   - [Window.onMetricsChanged](https://api.flutter.dev/flutter/dart-ui/Window/onMetricsChanged.html)
+  @override
+  void didChangeMetrics() {
+    // Should this method push a new scope with a new Device Context?
+    final window = WidgetsBinding.instance.window;
+    hub.addBreadcrumb(Breadcrumb(
+      message: 'Screen sized changed',
+      category: 'ui.lifecycle',
+      data: <String, dynamic>{
+        'new_pixel_ratio': window.devicePixelRatio,
+        'new_height': window.physicalSize.height,
+        'new_width': window.physicalSize.width,
+      },
+    ));
+  }
+
+  /// See also:
+  ///   - [Window.onPlatformBrightnessChanged](https://api.flutter.dev/flutter/dart-ui/Window/onPlatformBrightnessChanged.html)
+  @override
+  void didChangePlatformBrightness() {
+    final brightness = WidgetsBinding.instance.window.platformBrightness;
+    final brightnessDescription =
+        brightness == Brightness.dark ? 'dark' : 'light';
+
+    hub.addBreadcrumb(Breadcrumb(
+      message: 'Platform brightness was changed to $brightnessDescription.',
+      category: 'ui.lifecycle',
+    ));
+  }
+
+  /// See also:
+  ///   - [Window.onTextScaleFactorChanged]https://api.flutter.dev/flutter/dart-ui/Window/onTextScaleFactorChanged.html)
+  @override
+  void didChangeTextScaleFactor() {
+    final newTextScaleFactor = WidgetsBinding.instance.window.textScaleFactor;
+    hub.addBreadcrumb(Breadcrumb(
+      message: 'Text scale factor changed to $newTextScaleFactor.',
+      category: 'ui',
+      level: SentryLevel.warning,
```

Comment:
same as https://github.com/getsentry/sentry-dart/pull/203/files#r532241561

plus, why is this a `warning` level?